src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts

```diff
@@ -4,52 +4,152 @@
 import { Registry } from 'vs/platform/registry/common/platform';
-import { NotebookProviderRegistration, INotebookProviderRegistry, Extensions } from 'sql/workbench/services/notebook/common/notebookRegistry';
-import { MockContextKeyService } from 'vs/platform/keybinding/test/common/mockKeybindingService';
+import { IWorkspaceStorageChangeEvent, StorageScope } from 'vs/platform/storage/common/storage';
+import { ExtensionManagementService } from 'vs/workbench/services/extensionManagement/common/extensionManagementService';
+import { TestFileService, TestLifecycleService } from 'vs/workbench/test/browser/workbenchTestServices';
+import { TestExtensionService, TestStorageService } from 'vs/workbench/test/common/workbenchTestServices';
+
+/**
+ * class to mock azdata.nb.ServerManager object
+ */
+class TestServerManager implements azdata.nb.ServerManager {
+	isStarted: boolean = true; //by default our mock creates ServerManager in started mode.
+	onServerStarted: Event<void>;
+	async startServer(kernelSpec: azdata.nb.IKernelSpec): Promise<void> {
+		this.isStarted = true;
+	}
+	async stopServer(): Promise<void> {
+		this.isStarted = false;
+	}
+
+}
+
+/**
+ * TestNotebookManager - creates a NotebookManager object that helps keep track of state needed by testing
+ */
+class TestNotebookManager implements INotebookManager {
+	contentManager: undefined;
+	sessionManager: undefined;
+	constructor(
+		public providerId: string = 'providerId1',
+		public serverManager: TestServerManager = new TestServerManager()
+	) { }
+}
+
+/**
+ * * TestNotebookProvider - creates a NotebookManager object that helps keep track of state needed by testing
+ */
+class TestNotebookProvider implements INotebookProvider {
+	constructor(
+		public providerId: string = 'providerId1',
+		public manager: TestNotebookManager = new TestNotebookManager(providerId)
+	) { }
+
+	getNotebookManager(uri: URI): Thenable<INotebookManager> {
+		return Promise.resolve(this.manager);
+	}
 
-suite('Notebook Service Tests', function (): void {
-	let notebookService: INotebookService;
+	handleNotebookClosed(_notebookUri: URI): void {
+		// do nothing
+	}
+}
 
-	let installEvent: Emitter<InstallExtensionEvent>,
-		didInstallEvent: Emitter<DidInstallExtensionEvent>,
-		uninstallEvent: Emitter<IExtensionIdentifier>,
-		didUninstallEvent: Emitter<DidUninstallExtensionEvent>;
+suite('Test class ProviderDescriptor:', () => {
+	test('Verifies varies getter setters of Provider Descriptor', async () => {
+		const notebookProvider = <INotebookProvider>{};
+		const providerDescriptor = new ProviderDescriptor(notebookProvider);
+		assert.strictEqual(providerDescriptor.instance, notebookProvider, `providerDescriptor instance should be the value passed into the constructor`);
+		const providerInstancePromise = providerDescriptor.instanceReady;
+		assert.notStrictEqual(providerInstancePromise, undefined, `providerDescriptor instanceReady should return an undefined promise object`);
```

Comment:
You say it shouldn't return an undefined object but you're checking that it's NOT an undefined object

Reply:
will fix the comment

Reply:
fixed